cpp/oneapi/dal/algo/kmeans_init/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -14,93 +14,65 @@
 * limitations under the License.
 *******************************************************************************/
 
-#include <include/algorithms/kmeans/kmeans_init_types.h>
-#include <src/algorithms/kmeans/oneapi/kmeans_init_dense_batch_kernel_ucapi.h>
-
 #include "oneapi/dal/algo/kmeans_init/backend/gpu/compute_kernel.hpp"
-#include "oneapi/dal/algo/kmeans_init/backend/to_daal_method.hpp"
+#include "oneapi/dal/algo/kmeans_init/backend/gpu/compute_kernels_impl.hpp"
 #include "oneapi/dal/backend/interop/common_dpc.hpp"
 #include "oneapi/dal/backend/interop/error_converter.hpp"
-#include "oneapi/dal/backend/interop/table_conversion.hpp"
 #include "oneapi/dal/detail/error_messages.hpp"
 #include "oneapi/dal/table/row_accessor.hpp"
 
 namespace oneapi::dal::kmeans_init::backend {
 
-using std::int64_t;
 using dal::backend::context_gpu;
 
-namespace daal_kmeans_init = daal::algorithms::kmeans::init;
 namespace interop = dal::backend::interop;
+namespace pr = dal::backend::primitives;
+
+using result_t = compute_result<task::init>;
+
+using descriptor_t = detail::descriptor_base<task::init>;
+
+using input_t = compute_input<task::init>;
 
 template <typename Float, typename Method>
-using daal_kmeans_init_kernel_t =
-    daal_kmeans_init::internal::KMeansInitDenseBatchKernelUCAPI<to_daal_method<Method>::value,
-                                                                Float>;
-
-template <typename Float, typename Method, typename Task>
-static compute_result<Task> call_daal_kernel(const context_gpu& ctx,
-                                             const detail::descriptor_base<Task>& params,
-                                             const table& data) {
+static result_t call_daal_kernel(const context_gpu& ctx,
+                                 const descriptor_t& params,
+                                 const table& data) {
     auto& queue = ctx.get_queue();
     interop::execution_context_guard guard(queue);
 
     const int64_t column_count = data.get_column_count();
+    const int64_t row_count = data.get_row_count();
     const int64_t cluster_count = params.get_cluster_count();
 
-    daal_kmeans_init::Parameter par(dal::detail::integral_cast<std::size_t>(cluster_count));
-
-    const auto daal_data = interop::convert_to_daal_table(queue, data);
+    auto data_ptr =
+        row_accessor<const Float>(data).pull(queue, { 0, -1 }, sycl::usm::alloc::device);
+    auto arr_data = pr::ndarray<Float, 2>::wrap(data_ptr, { row_count, column_count });
```

Comment:
We have function called `table2ndarray` in `primitives/utils.hpp`

Reply:
Replaced